cmd/incus/completion.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func (g *cmdGlobal) cmpInstances(toComplete string) ([]string, cobra.ShellCompDirective) {
+	results := []string{}
+
+	for remote, rc := range g.conf.Remotes {
+		if rc.Protocol != "incus" && rc.Protocol != "" {
+			continue
+		}
+
+		remoteToParse := toComplete
+
+		if strings.Contains(toComplete, ":") {
+			if remote != strings.Split(toComplete, ":")[0] {
+				continue
+			}
+		} else {
+			if toComplete != "" && remote != "local" {
```

Comment:
Global isn't always the default remote, so this probably should use `c.global.conf.DefaultRemote` instead.